src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -304,6 +299,11 @@ public DependencyContext Build(string[] userRuntimeAssemblies = null)
                 runtimeLibraries.Add(runtimeLibrary);
             }
 
+            if (_includeMainProjectInDepsFile)
```

Comment:
Does the ordering matter or is it just for clarity? The real fix is the unique identifier to avoid the duplicate key when calling `ToDictionary`.

Reply:
I thought it needed to be added after the other libraries so that it would detect the conflict, but it turns out that `_usedLibraryNames` is initialized in the constructor.  So I've undone the reordering.